osx/ka-lite-remover.sh

```diff
@@ -0,0 +1,145 @@
+#
+
+#----------------------------------------------------------------------
+# Global Variables
+#----------------------------------------------------------------------
+KALITE_MONITOR="/Applications/KA-Lite-Monitor.app"
+KALITE="kalite"
+KALITE_PLIST="org.learningequality.kalite.plist"
+HOME_LAUNCH_AGENTS="$HOME/Library/LaunchAgents"
+ROOT_LAUNCH_AGENTS="/Library/LaunchAgents"
+LIBRARY_PLIST="$LAUNCH_AGENTS/$KALITE_PLIST"
+KALITE_EXECUTABLE_PATH="$(which $KALITE)"
+KALITE_RESOURCES="/Users/Shared/ka-lite"
+KALITE_USR_BIN_PATH="/usr/bin"
+KALITE_USR_LOCAL_BIN_PATH="/usr/local/bin"
+
+REMOVE_FILES_ARRAY=()
+
+#----------------------------------------------------------------------
+# Functions
+#----------------------------------------------------------------------
+function append() {
+    eval $1[\${#$1[*]}]=$2
+}
+
+
+function remove_files_initiator {
+
+    for file in "${REMOVE_FILES_ARRAY[@]}"; do
+        if [ -e "${file}" ]; then
+            echo "Now removing file: ${file}"
+            syslog -s -l error "Now removing file: ${file}"
+        fi
+    done
+
+    # Collect the directories and files to remove
+    if [ "$SCRIPT_NAME" != "preinstall" ]; then
+        osascript -e "do shell script \"/bin/rm -Rf ${REMOVE_FILES_ARRAY[*]}\" with administrator privileges"
+    else
+         sudo rm -Rf ${REMOVE_FILES_ARRAY[*]}
+    fi
+
+    for file in "${REMOVE_FILES_ARRAY[@]}"; do
+        if [ -e "${file}" ]; then
+            echo "An error must have occurred since a file that was supposed to be"
+            echo "removed still exists: ${file}"
+            syslog -s -l error "File still exists: ${file}"
+            echo ""
+            exit 1
+        fi
+    done
+}
+
+#----------------------------------------------------------------------
+# Script
+#----------------------------------------------------------------------
+ENV=$(env)
+syslog -s -l error "Packages pre-installation initialize with env:'\n'$ENV" 
+
+pushd `dirname $0` > /dev/null
+SCRIPTPATH=`pwd`
+popd > /dev/null
+
+# Collect the directories and files to remove
+test -d /Applications/KA-Lite                   && REMOVE_FILES_ARRAY+=("/Applications/KA-Lite")
+test -f $SCRIPTPATH/KA-Lite_Uninstall.tool      && REMOVE_FILES_ARRAY+=("$SCRIPTPATH/KA-Lite_Uninstall.tool")
+test -d $KALITE_RESOURCES                       && REMOVE_FILES_ARRAY+=("$KALITE_RESOURCES")
+test -f $HOME_LAUNCH_AGENTS/$KALITE_PLIST       && REMOVE_FILES_ARRAY+=("$HOME_LAUNCH_AGENTS/$KALITE_PLIST")
+test -f $ROOT_LAUNCH_AGENTS/$KALITE_PLIST       && REMOVE_FILES_ARRAY+=("$ROOT_LAUNCH_AGENTS/$KALITE_PLIST")
+test -f $KALITE_USR_LOCAL_BIN_PATH/$KALITE      && REMOVE_FILES_ARRAY+=("$KALITE_USR_LOCAL_BIN_PATH/$KALITE")
+test -f $KALITE_USR_BIN_PATH/$KALITE            && REMOVE_FILES_ARRAY+=("$KALITE_USR_BIN_PATH/$KALITE")
+test -f $KALITE_MONITOR_APP                     && REMOVE_FILES_ARRAY+=("$KALITE_MONITOR_APP")
+
+if [ "$SCRIPT_NAME" != "preinstall" ]; then
```

Comment:
The `preinstall` is used in the packages env var.
I used that one to flag if the script is run by the packages or the uninstaller script.